test/hotspot/gtest/testutils.hpp

```diff
@@ -53,6 +53,8 @@ class GtestUtils : public AllStatic {
 // Convenience asserts
 #define ASSERT_NOT_NULL(p)  ASSERT_NE(p2i(p), 0)
 #define ASSERT_NULL(p)      ASSERT_EQ(p2i(p), 0)
+#define EXPECT_NOT_NULL(p)  EXPECT_NE(p2i(p), 0)
+#define EXPECT_NULL(p)      EXPECT_EQ(p2i(p), 0)
```

Comment:
These aren't "convenience asserts" - should they be somewhere else? What does EXPECT_NULL actually do?

Reply:
They extend the existing `ASSERT_xxx` and `EXPECT_xxx` macros we get from gtest (in gtest.h). I cannot add them to gtest, of course, they have to live within our gtest wrapper code. This is as good a place as any IMO. Note we do similar things with other test helpers, e.g. threadHelper.inline.hpp.

Reply:
> I meant different place in the file, rather than under "convenience asserts" when they are not assertions. I remain ignorant as to what the behaviour of EXPECT_X(p) is actually meant to be ??

EXPECT_x and ASSERT_x mimic the assertion macros with the same name that the gtest suite exposes. This is gtest lingo. It is assumed here that the reader knows how gtest tests are written, in particular what ASSERT and EXPECT are and how they work.

I can make it more obvious with a clarifying comment.